grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -160,7 +165,12 @@ class StockMovementController {
 
     def removeStockMovement = {
         StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
-        if (stockMovement?.shipment?.currentStatus == ShipmentStatusCode.PENDING || !stockMovement?.shipment?.currentStatus) {
+        if (stockMovement?.requisition?.origin?.id != session.warehouse.id) {
+            flash.error = "You are not able to delete stock movement from a location other than origin."
+            if (params.show) {
+                return redirect(action: "show", id: params.id)
```

Comment:
redirect does not return anything so this might do weird things. let's do it like i suggested.
```
redirect(action: "show", id: params.id)
return
```